### ISSUE TITLE
FlexboxLayout does not report its preferred/min size to parent layouts

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#11936]
### Bug Description

When a FlexboxLayout is used as a child of another layout (HorizontalLayout, VerticalLayout, GridLayout, etc.), 
it appears to report a zero preferred-width and zero preferred-height to that parent. As a result, siblings end up at the same position and the layouts overlap. 
The FlexboxLayout's own children still render at their natural sizes, so the overlap is visible.

### Reproducible Code (if applicable)

```slint
// Reproducer for a FlexboxLayout sizing bug.

// Expected: the two FlexboxLayouts sit side by side inside the outer
// HorizontalLayout, separated by 24px of spacing.
// Actual: both FlexboxLayouts are placed at x=padding (they overlap), because
// a FlexboxLayout does not report a non-zero preferred / minimum width to
// its parent layout. The contents inside each FlexboxLayout still render at
// their own natural sizes, so the two stacks visually collide.
//
// Swapping the column FlexboxLayout for a VerticalLayout (and the row one for
// a HorizontalLayout) makes the layout correct, confirming the missing piece
// is FlexboxLayout's layout-info propagation.

import { Button } from "std-widgets.slint";

export component Foo inherits Window {
    preferred-width: 600px;
    preferred-height: 120px;
    background: #f5f5f5;

    //VerticalLayout {
        HorizontalLayout {
            alignment: start;
            // align-items: start;   // (unrelated, but uncomment to panic)
            spacing: 24px;
            padding: 12px;

            FlexboxLayout {
                flex-direction: column;
                spacing: 8px;
                Button { text: "Portrait"; }
                Button { text: "Landscape"; }
                Button { text: "Auto"; }
            }

            FlexboxLayout {
                flex-direction: row;
                spacing: 8px;
                Button { text: "Start"; }
                Button { text: "Center"; }
                Button { text: "End"; }
            }
        }

    //    Rectangle { background: white;}
    //}
}
```

### Environment Details

- Slint Version: Master (1.17)

### Product Impact

Was trying to implement the RadioGroup that needs to switch between horizontal and vertical orientation using FlexboxLayout, but can't get good results.

**Comment by dfaure-kdab** [comment]
Thanks for the testcase. The flexbox didn't report zero - a wrapping column-flexbox reported only its first column's width to the parent, so the sibling was placed over the wrapped second column. 

Fixed:
1. i-slint-core no longer stretches wrapped column items to the full container width (fixes interpreter)
2. the compiler feeds a nested column-flex its real cross size so it reports the correct wrapped width.

https://github.com/slint-ui/slint/pull/12129
